gym_solo/core/termination.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+
+class Termination(ABC):
+    @abstractmethod
+    def __init__(self):
+        """This is used to define the termination condition
+        """
+        pass
+    
+    @abstractmethod
+    def is_terminated(self):
+      """This should implement functionanlity that determines when an episode
+      should terminate
+      """
+      pass
+
+
+class TerminationFactory:
+  def __init__(self):
+    """Create a new Termination Factory. This way users can add multiple
+    different termination conditions and different ways to execute them to
+    signal the end of an episode
+    """
+    self._termination = []
+    self._is_or = True
+
+  def register_observation(self, term: Termination):
+    """Add a termination condittion used to evaluate when an episode
+    has ended.
+
+    Args:
+      term (Termination): Termination condition used to signal the 
+      end of an episode 
+    """
+    pass
+
+  def register_execution_type_as_or(self, is_or: bool = True):
+    """If isOr is true, then all the termination conditions are executed as or
+    conditional statements
+    """
+    pass
```

Comment:
I don't really understand the point of this function. also, the `is_or` functionality is for all of the operations (i.e. you can't mix `and` and `or` operators because we aren't building the context tree. As a short term hack, couldn't you just make an attribute and have the user set it when needed? It might be a little ugly, but like it's pretty niche functionality so I'm ok with it.

Reply:
I was trying to leave support for later but for now, a simple implementation will do the job